lib/Traits/Configuration/ConfigParametersTraits.ts

```diff
@@ -0,0 +1,403 @@
+    description: "Branding bar configuration"
+  })
+  brandBar: BrandBarTraits = new BrandBarTraits();
+
+  @primitiveTrait({
+    name: "Disable my location",
+    type: "boolean",
+    description:
+      'True to disable the "Centre map at your current location" button.'
+  })
+  disableMyLocation: boolean = false;
+
+  @primitiveTrait({
+    name: "Disable splitter",
+    type: "boolean",
+    description: "True to disable the splitter tool."
+  })
+  disableSplitter: boolean = false;
+
+  @primitiveTrait({
+    name: "Disable pedestrian mode",
+    type: "boolean",
+    description: "True to disable the pedestrian tool."
+  })
+  disablePedestrianMode: boolean = false;
+
+  @primitiveTrait({
+    name: "Experimental features",
+    type: "boolean",
+    description: "True to enable experimental features."
+  })
+  experimentalFeatures: boolean = false;
+
+  @anyTrait({
+    name: "MagdaReferenceHeaders",
+    description: "True to disable the pedestrian tool."
+  })
+  magdaReferenceHeaders?: MagdaReferenceHeaders;
+
+  @primitiveArrayTrait({
+    name: "Location search bounding box",
+    type: "number",
+    description: "True to disable the pedestrian tool."
+  })
+  locationSearchBoundingBox?: number[];
+
+  @primitiveTrait({
+    name: "Google analytics key",
+    type: "boolean",
+    description:
+      "A Google API key for [Google Analytics](https://analytics.google.com).  If specified, TerriaJS will send various events about how it's used to Google Analytics.",
+    isNullable: true
+  })
+  googleAnalyticsKey?: string;
+
+  @anyTrait({
+    name: "Google analytics options",
+    description: "Google analytics options."
+  })
+  googleAnalyticsOptions?: any;
+
+  @objectTrait({
+    name: "Error service",
+    type: ErrorServiceOptionsTraits,
+    description: "Error service provider configuration."
+  })
+  errorService?: ErrorServiceOptionsTraits = undefined;
+
+  @anyTrait({
+    name: "Global disclaimer",
+    description: ""
+  })
+  globalDisclaimer: any;
+
+  @objectTrait({
+    name: "Welcome message",
+    type: WelcomeMessageTraits,
+    description: "Welcome message definition."
+  })
+  welcomeMessage: WelcomeMessageTraits = new WelcomeMessageTraits();
+
+  // TODO: make themeing TS
+  @anyTrait({
+    name: "Theme",
+    description:
+      'An object used to override theme properties - for example `{"logoHeight": "70px"}`.T his is applied in StandardUserInterface and merged in order of highest priority: `StandardUserInterface.jsx` `themeOverrides` prop -> `theme` config parameter (this object) -> default `terriaTheme` (see `StandardTheme.jsx`)'
+  })
+  theme: any = {};
+
+  @primitiveTrait({
+    name: "Show in app guides",
+    type: "boolean",
+    description: "True to display in-app guides."
+  })
+  showInAppGuides?: boolean = false;
+
+  @objectArrayTrait({
+    type: HelpItemTraits,
+    name: "Help content items",
+    description: "The content to be displayed in the help panel.",
+    idProperty: "index"
```

Comment:
Just dropping a comment here for a future full review: I think this should be:
```suggestion
    idProperty: "itemName",
    isRemovable: true
```
though I haven't thought through the consequences of either `idProperty` choices.

Reply:
I think `itemName` is good - then makes it easy to remove specific default help items